src/main/java/com/prajina/academy/dao/AbstractDAO.java

```diff
@@ -0,0 +1,15 @@
+package com.prajina.academy.dao;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+ 
+public abstract class AbstractDAO {
```

Comment:
just a note that we don't use the abstractDAO pattern on project.  no change is needed though.

Reply:
I think that the @Repository annotation on the DaoImpls  might take care of the session.